Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XRTests.cs

```diff
@@ -1,29 +1,648 @@
 using NUnit.Framework;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using UnityEngine;
 using UnityEngine.Experimental.Input;
-
+using UnityEngine.Experimental.Input.Utilities;
+using UnityEngine.Experimental.Input.Plugins.XR;
+using UnityEngine.Experimental.Input.Controls;
+
 public class XRTests : InputTestFixture
 {
-    /*
-    ////TODO: make the same kind of functionality work for aliases
+    [Test]
+    [Category("Layouts")]
+    public void Layouts_GenericDeviceCreatesLayoutThatExtendsXRHMD()
+    {
+        testRuntime.ReportNewInputDevice(
+            new InputDeviceDescription
+        {
+            interfaceName = XRUtilities.kXRInterface,
+            product = "XRGenericDevice",
+            manufacturer = "XRManufacturer",
+            capabilities = new XRDeviceDescriptor
+            {
+                deviceRole = DeviceRole.Generic,
+                inputFeatures = new List<XRFeatureDescriptor>()
+                {
+                    new XRFeatureDescriptor()
+                    {
+                        name = "SimpleFeature",
+                        featureType = FeatureType.Binary
+                    }
+                }
+            }.ToJson()
+        }.ToJson());
+
+        InputSystem.Update();
+
+        var generatedLayout = InputSystem.TryLoadLayout("XRInput::XRManufacturer::XRGenericDevice");
+        Assert.That(generatedLayout, Is.Not.EqualTo(null));
+        Assert.That(generatedLayout.extendsLayout, Is.EqualTo("XRHMD"));
+    }
+
+    [Test]
+    [Category("Devices")]
+    public void Devices_GenericDeviceRoleCreatesHMDDevice()
+    {
+        testRuntime.ReportNewInputDevice(
+            new InputDeviceDescription
+        {
+            interfaceName = XRUtilities.kXRInterface,
+            product = "XRGenericDevice",
+            manufacturer = "XRManufacturer",
+            capabilities = new XRDeviceDescriptor
+            {
+                deviceRole = DeviceRole.Generic,
+                inputFeatures = new List<XRFeatureDescriptor>()
+                {
+                    new XRFeatureDescriptor()
+                    {
+                        name = "SimpleFeature",
+                        featureType = FeatureType.Binary
+                    }
+                }
+            }.ToJson()
+        }.ToJson());
```

Comment:
Might be worth building something equivalent to `HID.HIDDeviceDescriptorBuilder` for XR.

Reply:
Playing around with the XRDeviceDescriptorBuilder, and I'm finding it's about the same amount of lines to build, but now needs some whole other class.  I'm thinking of holding off on that one for now, since testing is the only time I'd really create a XRDeviceDescriptor in C#.